resource/conf.go

```diff
@@ -8,8 +8,35 @@ type Conf struct {
 	// no default page size is set resulting in no pagination if no `limit` parameter
 	// is provided.
 	PaginationDefaultLimit int
+	// ForceTotal controls how total number of items on list request is computed.
+	// By default (User), if the total cannot be computed by the storage andler for
+	// free, no total metadata is returned until the user explicitely request it
+	// using the total=1 query-string parameter. Note that if the storage cannot
+	// compute the total and does not implement the resource.Counter interface,
+	// a "not implemented" error is returned.
+	//
+	// The Always mode always force the computation of the total (make sure the
+	// storage either compute the total on Find or implement the resource.Counter
+	// interface.
+	//
+	// Denied prevents the user from requesting the total.
+	ForceTotal ForceTotalMode
 }
 
+// ForceTotalMode defines Conf.ForceTotal modes
+type ForceTotalMode int
+
+const (
+	// User let the end-user request the total number of items on list requests
+	// using the total=1 query-string parameter
+	User ForceTotalMode = iota
+	// Always always force the total number of items on list requests
+	Always
```

Comment:
> I just can't see a use-case where the user would like to opt-out, do you see one?

You are right, I also fail to see any _strong_ use-case for it. I see a _weak_ one, but let's not worry about that.

Still, it would be nice to rename `User` to avoid confusion and perhaps also add a `Mode` prefix (ala `http.Status...`). I was quite confused the first time I read "By default (User)". It's not completely obvious what it means... Also remember that it will appear as `resource.User` in auto-completion and code, which could even lead people to believe it's a user object.

Would `ModeOptIn`, `ModeAlways`, `ModeDenied` be better? This even allows `ModeOptOut` to be added later if anyone actually requests it.